pkg/api/types.go

```diff
@@ -204,8 +224,10 @@ type AgentPoolProfile struct {
 	Subnet              string `json:"subnet"`
 	IPAddressCount      int    `json:"ipAddressCount,omitempty"`
 
-	FQDN             string            `json:"fqdn,omitempty"`
-	CustomNodeLabels map[string]string `json:"customNodeLabels,omitempty"`
+	FQDN                  string            `json:"fqdn,omitempty"`
+	CustomNodeLabels      map[string]string `json:"customNodeLabels,omitempty"`
+	PreprovisionExtension *Extension        `json:"preprovisionExtension"`
```

Comment:
nit: change to `PreProvisionExtension` (capital `P` in `Provision`)

(and `preProvisionExtension` for the json representation)